mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/quota/CassandraQuotaLimitDaoTest.java

```diff
@@ -0,0 +1,91 @@
+
+    @RegisterExtension
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraModule.aggregateModules(CassandraQuotaModule.MODULE));
+
+    @BeforeEach
+    private void setup() {
+        cassandraQuotaLimitDao = GuiceUtils.testInjector(cassandraCluster.getCassandraCluster()).getInstance(CassandraQuotaLimitDao.class);
+    }
+
+    @Test
+    void getQuotaLimitsShouldGetSomeQuotaLimitsSuccessfully() {
+        QuotaLimit expectedOne = QuotaLimit.builder().quotaComponent(QuotaComponent.MAILBOX).quotaScope(QuotaScope.DOMAIN).identifier("A").quotaType(QuotaType.COUNT).quotaLimit(200l).build();
+        QuotaLimit expectedTwo = QuotaLimit.builder().quotaComponent(QuotaComponent.MAILBOX).quotaScope(QuotaScope.DOMAIN).identifier("A").quotaType(QuotaType.SIZE).quotaLimit(100l).build();
+        cassandraQuotaLimitDao.setQuotaLimit(expectedOne).block();
+        cassandraQuotaLimitDao.setQuotaLimit(expectedTwo).block();
+
+        assertThat(cassandraQuotaLimitDao.getQuotaLimits(QuotaComponent.MAILBOX, QuotaScope.DOMAIN, "A").collectList().block())
+            .containsExactlyInAnyOrder(expectedOne, expectedTwo);
+    }
+
+    @Test
+    void setQuotaLimitShouldSaveObjectSuccessfully() {
+        QuotaLimit expected = QuotaLimit.builder().quotaComponent(QuotaComponent.MAILBOX).quotaScope(QuotaScope.DOMAIN).identifier("A").quotaType(QuotaType.COUNT).quotaLimit(100l).build();
+        cassandraQuotaLimitDao.setQuotaLimit(expected).block();
+
+        assertThat(cassandraQuotaLimitDao.getQuotaLimit(CassandraQuotaLimitDao.QuotaLimitKey.of(QuotaComponent.MAILBOX, QuotaScope.DOMAIN, "A", QuotaType.COUNT)).block())
+            .isEqualTo(expected);
+    }
+
+    @Test
+    void setQuotaLimitShouldSaveObjectSuccessfullyWhenLimitIsMinusOne() {
+        QuotaLimit expected = QuotaLimit.builder().quotaComponent(QuotaComponent.MAILBOX).quotaScope(QuotaScope.DOMAIN).identifier("A").quotaType(QuotaType.COUNT).quotaLimit(-1l).build();
+        cassandraQuotaLimitDao.setQuotaLimit(expected).block();
+
+        assertThat(cassandraQuotaLimitDao.getQuotaLimit(CassandraQuotaLimitDao.QuotaLimitKey.of(QuotaComponent.MAILBOX, QuotaScope.DOMAIN, "A", QuotaType.COUNT)).block())
+            .isEqualTo(expected);
+    }
+
+    @Test
+    void deleteQuotaLimitShouldDeleteObjectSuccessfully() {
+        QuotaLimit quotaLimit = QuotaLimit.builder().quotaComponent(QuotaComponent.MAILBOX).quotaScope(QuotaScope.DOMAIN).identifier("A").quotaType(QuotaType.COUNT).quotaLimit(100l).build();
+        cassandraQuotaLimitDao.setQuotaLimit(quotaLimit).block();
+        cassandraQuotaLimitDao.deleteQuotaLimit(CassandraQuotaLimitDao.QuotaLimitKey.of(QuotaComponent.MAILBOX, QuotaScope.DOMAIN, "A", QuotaType.COUNT)).block();
+
+        assertThat(cassandraQuotaLimitDao.getQuotaLimit(CassandraQuotaLimitDao.QuotaLimitKey.of(QuotaComponent.MAILBOX, QuotaScope.DOMAIN, "A", QuotaType.COUNT)).block())
+            .isNull();
+    }
+
```

Comment:
What happens if I try to set quota limit to -100?

Reply:
No maybe it should be handled on a upper layer, as with mailbox quotas. Sorry :)